File: Java/1725_Number_Of_Rectangles_That_Can_Form_The_Largest_Square.java
class Solution {
    public int countGoodRectangles(int[][] rectangles) {
        int[] lengths = new int[rectangles.length];
        int result = 0;
        int largest = Integer.MIN_VALUE;
        
        for(int i = 0; i < rectangles.length; i++){
            lengths[i] = Math.min(rectangles[i][0], rectangles[i][1]);
            if(lengths[i] > largest){
                result = 1;
                largest = lengths[i];
            }
            else if(lengths[i] == largest){
                result++;
            }
        }
        return result;
    }
}

File: CS/1725_Number_Of_Rectangles_That_Can_Form_The_Largest_Square.cs
public class Solution {
    public int CountGoodRectangles(int[][] rectangles) {
        int result = 0;
        int temp = 0;
        var dic = new Dictionary<int, int>();
        foreach(var rectangle in rectangles){
            int i = Math.Min(rectangle[0], rectangle[1]);
            if(!dic.ContainsKey(i)){
                dic.Add(i, 0);
            }
            dic[i] += 1;
            if(temp <= i){
                temp = i;
                result = dic[i];
            }
        }
        
        return result;
    }
}
